File: pset_2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int get_letters(string input);
int get_words(string input);
int get_sentences(string input);

int main(void)
{
    // getting user input
    string text = get_string("Text: ");

    // calculating # of letters
    int letters = get_letters(text);
    // calculating # of words
    int words = get_words(text);
    // calculating # of sentences
    int sentences = get_sentences(text);

    // computing grade level
    float L = ((float) letters / (float) words) * 100;
    float S = ((float) sentences / (float) words) * 100;
    int grade_level = round(0.0588 * L - 0.296 * S - 15.8);

    // deciding what to print based on grade level
    if (grade_level < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade_level >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) grade_level);
    }
}

// function for getting the # of letters
int get_letters(string input)
{
    int num_letters = 0;
    for (int i = 0, n = strlen(input); i < n; i++)
    {
        if (isalpha(input[i]))
        {
            num_letters++;
        }
    }
    return num_letters;
}

// function for getting the # of words
int get_words(string input)
{
    int num_words = 0;
    for (int i = 0, n = strlen(input); i < n; i++)
    {
        if (isblank(input[i]))
        {
            num_words++;
        }
    }
    num_words++;
    return num_words;
}

// function for getting the # of sentences
int get_sentences(string input)
{
    int num_sentences = 0;
    for (int i = 0, n = strlen(input); i < n; i++)
    {
        if (input[i] == '!' || input[i] == '.' || input[i] == '?')
        {
            num_sentences++;
        }
    }
    return num_sentences;
}

File: pset_6/sentimental-readability/readability.py
import cs50

def main():
    text = input("Text: ")

    # calculating number of letters
    letters = get_letters(text)
    # calculating number of words
    words = get_words(text)
    #calculating number of sentences
    sentences = get_sentences(text)

    # computing grade level
    L = (letters / words) * 100
    S = (sentences / words) * 100
    grade_level = round(0.0588 * L - 0.296 * S - 15.8);

    # deciding what to print based on grade level
    if grade_level < 1:
        print("Before Grade 1")
    elif grade_level >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {int(grade_level)}")

# function for getting the # of letters
def get_letters(input):
    num_letters = 0
    for i in range(len(input)):
        if (input[i]isalpha()):
            num_letters += 1
    return num_letters

# function for getting the # of words
def get_words(input):
    num_words = 0
    for i in range(len(input)):
        if (input[i].isblank()):
            num_words += 1
    num_words += 1
    return num_words

# function for getting the # of sentences
def get_sentences(input):
    num_sentences = 0
    for i in range(len(input)):
        if (input[i] == '!' or input[i] == '.' or input[i] == '?'):
            num_sentences += 1;
    return num_sentences

main()
